cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -150,31 +156,43 @@ CacheAllocator<CacheTrait>::~CacheAllocator() {
 }
 
 template <typename CacheTrait>
-std::unique_ptr<MemoryAllocator>
-CacheAllocator<CacheTrait>::createNewMemoryAllocator() {
+ShmSegmentOpts CacheAllocator<CacheTrait>::createShmCacheOpts() {
+  if (memoryTierConfigs.size() > 1)
+    throw std::invalid_argument("CacheLib only supports a single memory tier");
+
   ShmSegmentOpts opts;
   opts.alignment = sizeof(Slab);
-  opts.typeOpts = PosixSysVSegmentOpts(config_.usePosixShm);
+
+  if (memoryTierConfigs.size()) {
```

Comment:
please explain this size check - I didn't understand why size > 0 means there'll be a file shm segment

Reply:
Is this for backward compatibility reasons ?